app/rest/metrics-collector/src/main/java/io/syndesis/rest/metrics/collector/MetricsCollector.java

```diff
@@ -0,0 +1,318 @@
+        Optional<Date> startDate = Optional.empty(); //we may have no more live pods for this integration
+
+        for (RawMetrics raw:metrics.values()) {
+            messages += raw.getMessages();
+            errors += raw.getErrors();
+            //Let's simply grab the oldest living pod, we will need to revisit when doing rolling upgrades etc
+            if (livePodIds.contains(raw.getPod())) {
+                if (startDate.isPresent()) {
+                    if (raw.getStartDate().get().before(startDate.get())) {
+                        startDate = raw.getStartDate();
+                    }
+                } else {
+                    startDate = raw.getStartDate();
+                }
+            }
+            if (raw.getLastProcessed().isPresent()) {
+                if (lastProcessed.isPresent()) {
+                    lastProcessed = raw.getLastProcessed().get().after(lastProcessed.get()) ? raw.getLastProcessed() : lastProcessed;
+                } else {
+                    lastProcessed = raw.getLastProcessed();
+                }
+            }
+        }
+
+        return new IntegrationMetricsSummary.Builder()
+                .id(integrationId)
+                .messages(messages)
+                .errors(errors)
+                .start(startDate)
+                .lastProcessed(lastProcessed)
+                .build();
+    }
+
+    /**
+     * Adds the RawMetrics of dead pods to a special HISTORY bucket. Each
+     * Integration should only have 1 HISTORY bucket and 1 bucket per live
+     * pod.
+     *
+     * @param integrationId
+     * @param metrics
+     * @param livePodIds
+     * @throws IOException
+     */
+    /*default*/ void curateDeadPodMetrics(
+            String integrationId,
+            String integration,
+            Map<String,RawMetrics> metrics,
+            Set<String> livePodIds) throws IOException {
+
+        for (Map.Entry<String, RawMetrics> entry : metrics.entrySet()) {
+            if (! entry.getKey().equals(HISTORY) && ! livePodIds.contains(entry.getKey())) { //dead pod check
+                if (metrics.containsKey(HISTORY)) {
+                    //add to existing history
+                    RawMetrics history = metrics.get(HISTORY);
+                    RawMetrics dead = entry.getValue();
+                    Date lastProcessed = history.getLastProcessed().orElse(new Date(0)).after(dead.getLastProcessed().orElse(new Date(0)))
+                            ? history.getLastProcessed().orElse(null) : dead.getLastProcessed().orElse(null);
+                    RawMetrics updatedHistoryMetrics = new RawMetrics.Builder()
+                            .integrationId(integrationId)
+                            .integration(integration)
+                            .pod(history.getIntegrationId() + ":" + dead.getPod())
+                            .messages(history.getMessages() + dead.getMessages())
+                            .errors(history.getErrors() + dead.getErrors())
```

Comment:
If you would not collect metrics per pod but per integration you would never have to do such 'curation', as the metrics 'per integration' is just incremented. After all that's what the user wants to see. I don't understand why you need to complicate this (or otherwise, I don't see the benefit on doing it that fine granuarly).